### PR TITLE
feat: more user friendly settings

### PR DESCRIPTION
**Pull request by ldez**
- set compat-mode to false by default.
- add `force experimental`
- `skipFrontMatter` become `frontMatter` : positive expression always better!
- add descriptions for options
- ordering options
- use enum for the safe mode definition
- change 'Show Table of Content' to 'Force show Table of Content', add explanations

capture du 2016-05-29 20-21-53

### CODE REVIEW

**Comment by anthonny** [comment]
It's really better 👍

File: lib/main.coffee [hunk 1]
```diff
@@ -7,38 +7,75 @@ module.exports =
...
+        Mutually exclusive with the notitle attribute.
+        '''
       type: 'boolean'
       default: true
+      order: 3
     safeMode:
+      description: '''
+        Set safe mode level: `unsafe`, `safe`, `server` or `secure`.
+
+        Disables potentially dangerous macros in source files, such as `include::[]`.
```
Comment (Mogztter) [inline lib/main.coffee]: Maybe we should explain how to allow include macros ?
"Set safe mode level to safe to enable include macros, but restricts access to ancestor paths of source file."

Comment (ldez) [inline lib/main.coffee]: In the next version of Atom, we will able to define a description for each value:

```coffee

    safeMode:
      description: '''
        Set safe mode level: `unsafe`, `safe`, `server` or `secure`.

        Disables potentially dangerous macros in source files, such as `include::[]`.

        http://asciidoctor.org/docs/user-manual/#running-asciidoctor-securely
        '''
      type: 'string'
      default: 'safe'
      enum: [
        value: 'unsafe',
        description: 'A safe mode level that disables any security features enforced by Asciidoctor.'
      ,
        value: 'safe',
        description: 'This safe mode level prevents access to files which reside outside of the parent directory of the source file. '
      ,
        value: 'server',
        description: 'A safe mode level that disallows the document from setting attributes that would affect the rendering of the document.'
      ,
        value: 'secure',
        description: 'A safe mode level that disallows the document from attempting to read files from the file system and including their contents into the document.'
      ]
```

**Comment by ldez** [comment]
I changed 'Show Table of Content' to 'Force show Table of Content', and added explanations

**Comment by ldez** [comment]
Contains a breaking change:
- rename options (add/delete)

version 2.0.0

**Comment by mojavelinux** [comment]
Overall, I like it.

I would prefer an action in front of "Front Matter". Perhaps "Allow Front Matter". (Technically, it just discards it, but it's also accurate to say it's being allowed).

I'm not sure how I feel about "Force" at the beginning of the title (Experimental Extensions and Table of Contents). It doesn't read well in English. Perhaps it would read better as "Enable experimental extensions (forced)" or "Always enable experimental extensions". I kind of like the second one better.

Can you make those updates?

**Comment by ldez** [comment]
@mojavelinux not problem :+1:

**Pull request merged**